src/hooks/useBridgedBlocks.ts

```diff
@@ -39,34 +39,26 @@ const useBridgedBlocks = ({ isApiReady, api, chain }: Props) => {
   const [bestBridgedFinalizedBlock, setBestBridgedFinalizedBlock] = useState('');
 
   const { bridgedGrandpaChain } = getSubstrateDynamicNames(chain);
-  useEffect(() => {
-    if (!api || !isApiReady || !chain) {
-      return;
-    }
+  useEffect((): (() => void) => {
+    const shouldProceed: boolean = !!(api && isApiReady && chain);
 
-    let unsubBestFinalized: () => void;
-    let unsubImportedHeaders: () => void;
+    let unsubImportedHeaders: Promise<VoidFn> | null;
+    const unsubBestFinalized: Promise<VoidFn> | null = shouldProceed
+      ? api.query[bridgedGrandpaChain].bestFinalized((res: CodecHeaderId) => {
+          const bestBridgedFinalizedBlock = res.toString();
+          unsubImportedHeaders = shouldProceed
+            ? api.query[bridgedGrandpaChain].importedHeaders(bestBridgedFinalizedBlock, (res: any) => {
+                if (res.toJSON()) {
+                  setBestBridgedFinalizedBlock(res.toJSON().number);
+                }
+              })
+            : null;
+        })
+      : null;
 
-    api.query[bridgedGrandpaChain]
-      .bestFinalized((res: CodecHeaderId) => {
-        const bestBridgedFinalizedBlock = res.toString();
-        api.query[bridgedGrandpaChain]
-          .importedHeaders(bestBridgedFinalizedBlock, (res: any) => {
-            if (res.toJSON()) {
-              setBestBridgedFinalizedBlock(res.toJSON().number);
-            }
-          })
-          .then((unsub) => {
-            unsubImportedHeaders = unsub;
-          });
-      })
-      .then((unsub) => {
-        unsubBestFinalized = unsub;
-      });
-
-    return function cleanup() {
-      unsubImportedHeaders && unsubImportedHeaders();
-      unsubBestFinalized && unsubBestFinalized();
+    return async (): Promise<void> => {
+      unsubImportedHeaders && (await unsubImportedHeaders)();
+      unsubBestFinalized && (await unsubBestFinalized)();
```

Comment:
```suggestion
      unsubBestFinalized && (await unsubBestFinalized)();
      unsubImportedHeaders && (await unsubImportedHeaders)();
```

With previous order you still introduce a race condition. IMHO it would be better to have only one promise which returns two unsubscribe functions, or yet even better - one promise, one function (to unsubscribe them all)!